### PR TITLE
Update Mock OpenAI Client 

### PR DESCRIPTION
**Pull request by LANNDS18**
## Description

 add mock class to simulate openai client behaviour, update test client to enable the argument checking

## Types of Changes

- [x] New feature (non-breaking change that adds functionality)

## Checklist

- [x] I have read the project's contributing guidelines.
- [x] My code follows the project's coding style.
- [x] I have written or updated unit tests for the new functionality.
- [x] I have updated the comments in code which is necessary,
- [x] I have rebased all the commit in my git.
- [x] I have updated the documentation as necessary.
- [x] I have added a descriptive commit message with a summary of the changes.

### CODE REVIEW

**Comment by BinLiang2021** [comment]
1. Line 48: The import statement should start from the root directory, like this: from xyz.utils.test_tool.mock_openai import MockOpenAI.
2. The comments and docstrings in test_client.py (e.g., lines 55, 79, 2-32) seem unchanged. Could you double-check to ensure they are updated appropriately?
3. There are some commented-out lines from 137 to 140. Is there a reason these lines haven't been deleted? If they are no longer needed, it might be cleaner to remove them.
4. The content after line 405 should not be in this file. Please move the unit tests to the test/xyz folder to maintain consistent directory structure with the xyz folder.
5. There appears to be an issue with the initialization requiring a logit_bias\n parameter. Why is this parameter mandatory? The absence of logit_bias leads to an error, which suggests that the default handling might need improvement.
6. There is currently no validation for an illegal model name during initialization, which does not trigger an error (other parameters being the same). I recommend adding a check both at initialization and during the request phase. Additionally, an illegal model name does not trigger an error during generation, which should be addressed to prevent runtime issues.

File: xyz/utils/test_tool/test_client.py [hunk 1]
```diff
@@ -32,19 +32,22 @@
...
 
 from dotenv import load_dotenv
-from openai import OpenAIError
 from openai import OpenAI
+from openai import OpenAIError
 from openai import Stream
 from openai.types.chat import ChatCompletion, ChatCompletionChunk
+from typing_extensions import Literal
+
+from mock_openai import MockOpenAI
```
Comment (BinLiang2021) [inline xyz/utils/test_tool/test_client.py:48]: Should be from xyz.utils.test_tool.mock_openai import MockOpenAI

File: xyz/utils/test_tool/test_client.py [hunk 2]
```diff
@@ -219,3 +400,43 @@ def stream_run(self, messages: list, images: list) -> Generator[str, None, None]
                 print(f"The error: {error_message}")
                 print(f"The messages: {messages}")
                 time.sleep(2)
+
+
+if __name__ == "__main__":
+    test_client = TestClient(api_key="sk-1", top_p=0.8911, n=2, logit_bias={"s": 0})
```
Comment (BinLiang2021) [inline xyz/utils/test_tool/test_client.py:406]: If I do not set n/logit_bias in the init method, it will raise error, why?

**Review by BinLiang2021 — CHANGES_REQUESTED** [review]

File: xyz/utils/test_tool/test_client.py (hunk 2 above)
Comment (LANNDS18) [reply xyz/utils/test_tool/test_client.py]: Bug-fix has been pushed

**Comment by LANNDS18** [comment]
Bug-fix has been pushed

**Comment by BinLiang2021** [comment]
1. Line 48: The import statement should start from the root directory, like this: from xyz.utils.test_tool.mock_openai import MockOpenAI. You did not fix this one.
2. why model name 'gpt-4-turbo' will raise an error?
3. In test_client.py line: 504. You did not import the time.

Could you please double check these error? And do the same check on the similar things.

**Pull request merged**